packages/hash/api/src/graph/knowledge/primitive/entity.ts

```diff
@@ -338,7 +339,7 @@ export const updateEntity: ImpureGraphFunction<
      *
      * @see https://app.asana.com/0/1201095311341924/1203285029221330/f
      * */
-    entityTypeId: entity.metadata.entityTypeId,
+    entityTypeId: entityTypeId ?? entity.metadata.entityTypeId,
```

Comment:
Logic here: If entityTypeId is passed to this function, use the passed entity type, else use entity's entity type (means it's unchanged)